taigit/src/libraries/Taiga.ts

```diff
@@ -877,4 +877,45 @@ check_for_retrospective(project_id : number, sprint_id : number) : Promise<wiki_
         }
     }
     return retro_page;
-}
\ No newline at end of file
+}
+
+
+
+**
+ * @summary This call return estimate of sprint plan need to finish the project (unit points) based on project Id
+ * @param projectId the ID for the access of project
+ * @returns
+ *  * Array of sprint plan [sprint_num, velocity]
+ * * {
+ *          sprint_num : number // num of sprint needed
+ *          velocity : number //estimate sprint velocity
+ * }
+ */
+
+export async function
+proj_spplan_res(projId : number) : Promise<Array<[number,number]>> {
+    let velocity_list : Array<[boolean, number]> =  (await proj_spvelocity(projId));
+    //test case//286226//306316
+    //calculate sprint velocity stats
+    let v_min : number = Number.MAX_VALUE;
+    let v_max : number = 0;
+    let v_avg : number = 0;
+    let v_sum : number = 0;
+    let n_s : number = 0;
+    for(let v of velocity_list){
+        if(v[0]) {
+            if (v[1] < v_min)
+                 v_min = v[1];
+             if (v[1] > v_max)
+                 v_max = v[1];
+             v_sum += v[1];
+             n_s += 1;
+         }
+
+    }
+    v_avg=v_sum/n_s;
+    //estimate future sprint planning
+    let project = (await axios.get("https://api.taiga.io/api/v1/projects/" + projId.toString() + '/stats')).data;
+    let rest_points = project.defined_points-project.closed_points;
+    return [[rest_points/v_min,v_min],[rest_points/v_avg,v_avg],[rest_points/v_max,v_max]];
```

Comment:
this should be an object not an array

Reply:
Since it is not a complex structure, I store it in array to read it faster for further use.

Reply:
It is a complex structure. In order to understand what any of the 6 numbers in the the list of lists are, I would need to go into your code and dissect it. They need labels so i can call `data.minumum.velocity` instead of `data[0][1]` where those numbers mean nothing to me.  

Also since when have we ever been speed conscious in this project and why would it apply now? Theres dozens to hundreds of other considerations to be made to increase the speed and efficiency of this project before you start worrying about the benefits of using an array over an object. You should definitely read up on [Premature Optimization](http://wiki.c2.com/?PrematureOptimization) and how it applies to Javascript specifically.

Reply:
If you see the comment of the API, you will see. @returns
 *  * Array of sprint plan [sprint_num, velocity]
 * * {
 *          sprint_num : number // num of sprint needed
 *          velocity : number //estimate sprint velocity
 * }

So in the array there are 3 elements. Each of them is a sprint plan.  each sprint plan is [ sprint_num, velocity]